src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -2770,6 +2949,264 @@ QgsProcessingParameterPoint *QgsProcessingParameterPoint::fromScriptCode( const
   return new QgsProcessingParameterPoint( name, description, definition, isOptional );
 }
 
+QgsProcessingParameterGeometry::QgsProcessingParameterGeometry( const QString &name, const QString &description,
+    const QVariant &defaultValue, bool optional, const QList<int> &geometryTypes )
+  : QgsProcessingParameterDefinition( name, description, defaultValue, optional ),
+    mGeomTypes( geometryTypes )
+{
+
+}
+
+QgsProcessingParameterDefinition *QgsProcessingParameterGeometry::clone() const
+{
+  return new QgsProcessingParameterGeometry( *this );
+}
+
+bool QgsProcessingParameterGeometry::checkValueIsAcceptable( const QVariant &input, QgsProcessingContext * ) const
+{
+  if ( !input.isValid() )
+    return mFlags & FlagOptional;
+
+  if ( input.canConvert<QgsProperty>() )
+  {
+    return true;
+  }
+
+  bool anyTypeAllowed = mGeomTypes.isEmpty() || mGeomTypes.contains( QgsWkbTypes::UnknownGeometry );
+
+  if ( input.canConvert< QgsGeometry >() )
+  {
+    return anyTypeAllowed || mGeomTypes.contains( input.value<QgsGeometry>().type() );
+  }
+
+  if ( input.canConvert< QgsReferencedGeometry >() )
+  {
+    return anyTypeAllowed || mGeomTypes.contains( input.value<QgsReferencedGeometry>().type() );
+  }
+
+  if ( input.canConvert< QgsPointXY >() )
+  {
+    return anyTypeAllowed || mGeomTypes.contains( QgsWkbTypes::PointGeometry );
+  }
+
+  if ( input.canConvert< QgsRectangle >() )
+  {
+    return anyTypeAllowed || mGeomTypes.contains( QgsWkbTypes::PolygonGeometry );
+  }
+
+  if ( input.canConvert< QgsReferencedPointXY >() )
+  {
+    return anyTypeAllowed || mGeomTypes.contains( QgsWkbTypes::PointGeometry );
+  }
+
+  if ( input.canConvert< QgsReferencedRectangle >() )
+  {
+    return anyTypeAllowed || mGeomTypes.contains( QgsWkbTypes::PolygonGeometry );
+  }
+
+  if ( input.type() == QVariant::String )
+  {
+    if ( input.toString().isEmpty() )
+      return mFlags & FlagOptional;
+  }
+
+  // Match against EWKT
+  QRegularExpression rx( QStringLiteral( "^\\s*(?:CRS=(.*);)?(.*)$" ) );
+
+  QRegularExpressionMatch match = rx.match( input.toString() );
+  if ( match.hasMatch() )
+  {
+    QgsGeometry g = QgsGeometry::fromWkt( match.captured( 2 ) );
+    if ( ! g.isNull() )
+    {
+      return anyTypeAllowed || mGeomTypes.contains( g.type() );
+    }
+    else
+    {
+      QgsMessageLog::logMessage( QObject::tr( "Error creating geometry: \"%1\"" ).arg( g.lastError() ), QObject::tr( "Processing" ) );
+    }
+  }
+  return false;
+}
+
+QString QgsProcessingParameterGeometry::valueAsPythonString( const QVariant &value, QgsProcessingContext &context ) const
+{
+  auto asPythonString = []( const QgsGeometry & g, QString authid = QString() )
+  {
+    if ( authid.isEmpty() )
+      return QStringLiteral( "'%1'" ).arg( g.asWkt() );
+    else
+      return QStringLiteral( "'CRS=%2;%1'" ).arg( g.asWkt(), authid );
```

Comment:
```suggestion
      return QgsProcessingUtils::stringToPythonLiteral( QStringLiteral( "CRS=%1;%2" ).arg( crs, c.authid().isEmpty() ? c.toWkt( QgsCoordinateReferenceSystem::WKT_PREFERRED ) : c.authid() ) );
```

Reply:
I do not understand the change, is there a mispelling error ?

Reply:
Yes, sorry -- fixed

Reply:
Thx. I think there should be `g.asWkt()` as second argument. So, the idea is to provide crs as WKT format if it is a custom CRS.

Reply:
Fixed, with `g.asWkt()` as second argument for `arg`